overlord/snapstate/snapstate_update_test.go

```diff
@@ -5264,11 +5273,9 @@ func (s *snapmgrTestSuite) TestUpdateManyWaitForBasesUC18(c *C) {
 		}
 	}
 
-	// Note that "core18" is not part of the prereqs for some-app
-	// as it does not use this base.
 	c.Check(prereqs, DeepEquals, map[string]bool{
 		"some-base": true,
-		"snapd":     true,
+		"core18":    true,
```

Comment:
why is snapd not longer in this?

Reply:
Super good question, the reason this has changed is because the dependency logic is simplified, instead of everything depending on multiple precursors, they now depend only on the last essential snap. So in this case we have two essential snaps updating:

snapd & core18 (boot-base). So the dependency graph currently goes like this:

```
snapd => core18 => some-base => some-snap
                            ============>  some-snap
```

Because currently it's set up so that all non-essential snaps depend on the last essential-snap, and all bases. The last essential-snap being core18. If core18 was not a part of this, then the last essential-snap would be snapd.

Before the dependency graph would look like

```
snapd => core18
           => some-base => some-snap
           ============> some-snap
```

So the primary change in this PR is that we are serializing the essential-snap updates, and making sure that boot-base/gadget/kernel are always transactional.

I think something similar to this write-up makes sense to include in the description for the PR (i'll update this).